diff --git a/node_modules/next/dist/server/base-server.js b/node_modules/next/dist/server/base-server.js
index a6577a6..7517512 100644
--- a/node_modules/next/dist/server/base-server.js
+++ b/node_modules/next/dist/server/base-server.js
@@ -1756,8 +1756,8 @@ class Server {
                     }
                 }
             }
-            if (this.minimalMode && isSSG && ((_cachedData_headers = cachedData.headers) == null ? void 0 : _cachedData_headers[_constants2.NEXT_CACHE_TAGS_HEADER])) {
-                res.setHeader(_constants2.NEXT_CACHE_TAGS_HEADER, cachedData.headers[_constants2.NEXT_CACHE_TAGS_HEADER]);
+            if (isSSG && ((_cachedData_headers = cachedData.headers) == null ? void 0 : _cachedData_headers[_constants2.NEXT_CACHE_TAGS_HEADER])) {
+                res.setHeader('Cache-Tag', cachedData.headers[_constants2.NEXT_CACHE_TAGS_HEADER]);
             }
             // If the request is a data request, then we shouldn't set the status code
             // from the response because it should always be 200. This should be gated
diff --git a/node_modules/next/dist/server/next.js b/node_modules/next/dist/server/next.js
index f570f6d..4934b11 100644
--- a/node_modules/next/dist/server/next.js
+++ b/node_modules/next/dist/server/next.js
@@ -101,6 +101,7 @@ class NextServer {
         return async (req, res, parsedUrl)=>{
             return (0, _tracer.getTracer)().trace(_constants2.NextServerSpan.getRequestHandler, async ()=>{
                 const requestHandler = await this.getServerRequestHandler();
+                res.setHeader("cache-tag", req.headers['x-invoke-path']);
                 return requestHandler(req, res, parsedUrl);
             });
         };
